semgrep/semgrep/core_runner.py

```diff
@@ -217,6 +217,72 @@ def _write_equivalences_file(self, fp: IO, equivalences: List[Equivalence]) -> N
         yaml.dump({"equivalences": [e.to_json() for e in equivalences]}, fp)
         fp.flush()
 
+    def _run_core_command(
+        self,
+        patterns_json: List[Any],
+        patterns: List[Pattern],
+        targets: List[Path],
+        language: Language,
+        rule: Rule,
+        rules_flag: str,
```

Comment:
I'd probably make this an enum instead of a `str` to avoid future potential bugs.

Reply:
Like this? https://docs.python.org/3.4/library/enum.html  
So it would be of type Mode and would evaluate to search = 1 and taint = 2 or something like that?

Reply:
I think `enum` is the wrong word for me to use.

I mean this:
```python
class Mode(str):
    pass

SearchMode = Mode("search")
TaintMode = Mode("taint")
```

Reply:
done